lizmap/modules/lizmap/lib/Logger/Item.php

```diff
@@ -0,0 +1,210 @@
+<?php
+/**
+ * Manage and give access to lizmap log item.
+ *
+ * @author    3liz
+ * @copyright 2012 3liz
+ *
+ * @see      http://3liz.com
+ *
+ * @license Mozilla Public License : http://www.mozilla.org/MPL/
+ */
+
+namespace Lizmap\Logger;
+
+use Lizmap\App;
+
+class Item
+{
+    // log item properties
+    private static $properties = array(
+        'label',
+        'logCounter',
+        'logDetail',
+        'logIp',
+        'logEmail',
+    );
+
+    // Log key
+    private $key = '';
+
+    private $data = array();
+
+    // log record keys
+    private static $recordKeys = array(
+        'key',
+        'user',
+        'content',
+        'repository',
+        'project',
+        'ip',
+        'email',
+    );
+
+    protected $context;
+
+    /**
+     * Construct the object, you should use the Log\Config::getLogItem() method
+     * which will call this constructor.
+     *
+     * @param string $key            the name of the item
+     * @param array  $readConfigPath the array containing the fields of lizmapLogConfig.ini.php
+     */
+    public function __construct($key, $readConfigPath, App\AppContextInterface $appContext)
+    {
+        $this->appContext = $appContext;
+
+        // Set each property
+        foreach (self::$properties as $property) {
+            if (isset($readConfigPath[$property])) {
+                $this->data[$property] = $readConfigPath[$property];
+            }
+        }
+
+        $this->key = $key;
+    }
+
+    /**
+     * Return log item key.
+     */
+    public function getKey()
+    {
+        return $this->key;
+    }
+
+    /**
+     * @return string[] list of properties name
+     */
+    public static function getSProperties()
+    {
+        return self::$properties;
+    }
+
+    /**
+     * @return string[] list of record keys
+     */
+    public function getRecordKeys()
+    {
+        return self::$recordKeys;
+    }
+
+    /**
+     * Return data for a log item.
+     *
+     * @param string $key Key of the log item
+     *
+     * @return mixed
+     */
+    public function getData($key)
+    {
+        if (!array_key_exists($key, $this->data)) {
+            return null;
+        }
+
+        return $this->data[$key];
+    }
+
+    /**
+     * Update the data for the log item in the ini file.
+     *
+     * @param mixed $data
+     *
+     * @return bool true if there were some modifications
+     */
+    public function update($data)
+    {
```

Comment:
what are these data? what data should contain? at least comments are missing. I would prefer some setter methods to set each data, and this method should only save $this->data. And i didn't find a usage of this method in the code. Probably it is not used at all.